### PR TITLE
DLS-7728 add update-change registered details page

### PR DESCRIPTION
**Pull request by GinnyThomas**
UI page: https://github.com/hmrc/soft-drinks-industry-levy-ui-tests/pull/154#issue-1841460250
Screenshot 2023-08-08 at 14 37 44
Screenshot 2023-08-08 at 14 37 38
Screenshot 2023-08-08 at 14 36 11

### CODE REVIEW

File: app/forms/updateRegisteredDetails/ChangeRegisteredDetailsFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import javax.inject.Inject
+
+import forms.mappings.Mappings
+import play.api.data.Form
+import play.api.data.Forms.set
+import models.updateRegisteredDetails.ChangeRegisteredDetails
+
+class ChangeRegisteredDetailsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[Set[ChangeRegisteredDetails]] =
```
Comment (ChrisHillCap) [inline app/forms/updateRegisteredDetails/ChangeRegisteredDetailsFormProvider.scala:28]: I think we need some form validation to check the isVoluntary flag as I think (its hard to read) the old code validated in the form
i..e
-  when im voluntary, i am allowed these checkboxes
- when im not voluntary, im not allowed to submit x checkbox

(taken from L300 in Variations journey in old fe
  ```  import ContactChangeType._
    val changeOptions =
      if (data.isVoluntary)
        ContactChangeType.values.filter(_ != ContactChangeType.Sites)
      else
        ContactChangeType.values

    for {
      change <- ask[Set[ContactChangeType]](
                 "change-registered-details",
                 validation = Rule.nonEmpty[Set[ContactChangeType]] alongWith Subset(changeOptions: _*)
               )```

File: it/controllers/updateRegisteredDetails/ChangeRegisteredDetailsControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+import pages.updateRegisteredDetails.ChangeRegisteredDetailsPage
+import play.api.http.HeaderNames
+import play.api.libs.json.Json
+import play.api.test.WsTestClient
+
+class ChangeRegisteredDetailsControllerISpec extends ControllerITTestHelper {
+
+  val normalRoutePath = "/"
+
+  "GET " + normalRoutePath - {
```
Comment (ChrisHillCap) [inline it/controllers/updateRegisteredDetails/ChangeRegisteredDetailsControllerISpec.scala:17]: can we check which checkboxes appear based on the "voluntary" flag for the GET AND POST(400) in the it tests

**Review by ChrisHillCap — CHANGES_REQUESTED** [review]
couple of comments, looks pretty swish apart from those

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

**Review by ChrisHillCap — APPROVED** [review]

**Comment by ChrisHillCap** [comment]
lgtm

**Comment by GinnyThomas** [comment]
Updates made per request - had difficulty with verified signature, new branch created as files were inadvertently lost.

**Pull request closed without merging**